### PR TITLE
[Task] Added a few more uitests for calling, including add and remove participant in the call

### PR DESCRIPTION
**Pull request by kevinyulu**
## Purpose

* Added a few more uitests for calling, including add and remove participant in the call

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [x] Other... Please describe: Ui test task

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Tests/AzureCommunicationUIDemoAppCallTests.swift [hunk 1]
```diff
@@ -68,4 +68,19 @@ class AzureCommunicationUIDemoAppCallTests: XCUITestBase {
         tapButton(accessibilityIdentifier: AccessibilityIdentifier.micAccessibilityID.rawValue)
         XCTAssertEqual(micButton.label, "Unmute")
     }
+
+    func testCallCompositeCallAddRemoteParticipantHandler() {
+        tapInterfaceFor(.callUIKit)
+        startExperience()
+
+        joinCall()
```
Comment (JoshuaLai) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Tests/AzureCommunicationUIDemoAppCallTests.swift:76]: Does the order of this make sense? We start the call the composite should be displayed, how would the UI test be able to tap the add participant button?

Comment (kevinyulu) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Tests/AzureCommunicationUIDemoAppCallTests.swift:76]: when the ui test starts there is an env variable to turn on a an overlay view (from demo app) which display a few buttons that can add mock participants. ui test then interact with these buttons and performs test after participant state changes.

**Review by vhuseinova-msft — APPROVED** [review]

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Mockings/CallingSDKEventsHandlerMocking.swift [hunk 2]
```diff
@@ -79,5 +81,74 @@ class CallingSDKEventsHandlerMocking: CallingSDKEventsHandler {
             self?.isRecordingActiveSubject.send(false)
         }
     }
+
+    func addParticipant() {
+        DispatchQueue.main.async { [weak self] in
```
Comment (emlynmac) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Mockings/CallingSDKEventsHandlerMocking.swift:86]: Would be nice to use the more modern `Task { }` syntax

Comment (vhuseinova-msft) [reply AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Mockings/CallingSDKEventsHandlerMocking.swift:86]: good catch! Will update for all funcs in the PR

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/FloatingUITestWindow.swift [hunk 3]
```diff
@@ -130,6 +142,55 @@ class FloatingUITestWindow: UIWindow {
...
         }
     }
+
+    @objc func addParticipantButtonTapped(sender: UIButton) {
+        debugPrint("UI Test:: AddParticipantButtonTapped")
+        Task {
+            do {
+                try await callingSDKWrapperMock?.addParticipant()
+            } catch {
+            }
```
Comment (emlynmac) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/Views/FloatingUITestWindow.swift:152]: Would be nice to catch the errors here (and everywhere else

**Review by emlynmac — APPROVED** [review]
Changes look good overall, some nit-pick issues.
Would be nice to run the tests against a few different device sets prior to merge

**Comment by vhuseinova-msft** [comment]
Created https://github.com/Azure/communication-ui-library-ios/pull/569 instead

**Pull request closed without merging**